timemachine/cpp/src/kernels/k_log_flat_bottom_bond.cuh

```diff
@@ -0,0 +1,107 @@
+#include "../fixed_point.hpp"
+#include "k_fixed_point.cuh"
+#include "k_flat_bottom_bond.cuh"
+
+template <typename RealType> RealType __device__ __forceinline__ stable_log_1_exp_neg(RealType x) {
+    const RealType LOG_2 = 0.693147180559945309417232121;
+    return x < LOG_2 ? log(-expm1(-x)) : log1p(-exp(-x));
+}
+
+template <typename RealType>
+void __global__ k_log_flat_bottom_bond(
+    const int B, // number of bonds
+    const double *__restrict__ coords,
+    const double *__restrict__ box,
+    const double *__restrict__ params, // [P, 3]
```

Comment:
No P defined here, is params actually [B, 3]?

Reply:
good catch - I'll fix this for both the flat_bottom and the harmonic_bond.